test/terra/backends/simulator_test_case.py

```diff
@@ -85,7 +85,13 @@ def _method_device(methods):
     # add special test device for cuStateVec if available
     cuStateVec = check_cuStateVec(available_devices)
 
-    gpu_methods = ["statevector", "density_matrix", "unitary", "tensor_network"]
+    gpu_methods = [
+        "statevector",
+        "density_matrix",
+        "unitary",
+        "tensor_network",
+        "matrix_product_state",
```

Comment:
I am not sure, if this is the correct way to do all the tests for MPS on GPU.
However, after this change I see a lot of MPS tests are being done on GPU, but its taking a lot of time :)

Reply:
![Screenshot_20240618_105918](https://github.com/Qiskit/qiskit-aer/assets/54737489/96edc19e-72e4-4be7-b840-56acd15986a3)

Looks like these tests are indeed running on GPU

For example this is the test:
`test.terra.backends.aer_simulator.test_cliffords.TestCliffords.test_cx_gate_nondeterministic_12___matrix_product_state____GPU__ [641.898773s] ... ok`

See the time,  its quiet long, thats why I am not sure if this is the correct place to get tests for MPS on GPU.

Any comment: @doichanj

Reply:
I also tested and it takes longer time than CPU. I think there are large overhead than simulating small qubits in the test cases, maybe we have to offload outer kernel of SVD to GPU to minimize overheads.

And line 114 should be modified to 
```
if cuStateVec and "tensor_network" != method and "matrix_product_state" != method:
```
to prevent enabling cuStateVec option

Reply:
Added suggestion at line 114

I also want to do some more modifications, and perhaps use directly the MPS methods from cuQuantum, but, today, I received a mail, stating that PR participating for unitaryHack should be merged till 19th June, to be eligible for the bounty.
That's why I tried to finish it today.

If you feel the PR is moderately good enough, and, I will do a follow up PR for any enhancements, then, I request you to merge this if and only if possible.

anyway, if the PR is not good enough, I will still do all that is required to get this PR upto the line, regardless of any bounty  :) 
@doichanj

Reply:
Unfortunately there is something wrong with CI test on MacOS and I can not merge this PR. I think it is not good to merge this PR until the performance issue will be solved. 
Anyway, could you write comment on #2112 that you will continue developing for this PR, that may support getting bounty for hack

Reply:
https://github.com/Qiskit/qiskit-aer/issues/2112#issuecomment-2176369574
Done! :)